bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java

```diff
@@ -582,6 +584,26 @@ public void run() {
                 return true;
             }
         });
+
+        // sheetView should always be anchored to the bottom of the screen
+        currentSheetViewHeight = sheetView.getMeasuredHeight();
+        sheetViewOnLayoutChangeListener = new OnLayoutChangeListener() {
+            @Override
+            public void onLayoutChange(View sheetView, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
+                int newSheetViewHeight = sheetView.getMeasuredHeight();
+                if (state != State.HIDDEN) {
+                    if (newSheetViewHeight < currentSheetViewHeight) {
+                        // The sheet can no longer be in the expanded state if it has shrunk
+                        if (state == State.EXPANDED) {
+                            setState(State.PEEKED);
+                        }
+                        setSheetTranslation(sheetView.getMeasuredHeight());
+                    }
+                }
```

Comment:
Also, it seems to me that the ```setSheetTranslation``` call at the end of this block leaves the bottomsheet in an inconsistent state: we set the state as peeked, but then set the translation to be the measured height (which is supposed to be the expanded state).